frontend/src/metabase/collections/components/BaseItemsTable.tsx

```diff
@@ -13,72 +17,84 @@ import type {
   OnToggleSelectedWithItem,
 } from "../types";
 
+import type { ActionMenuProps } from "./ActionMenu/ActionMenu";
 import {
-  BulkSelectWrapper,
   ColumnHeader,
-  LastEditedByCol,
   SortingControlContainer,
   SortingIcon,
   Table,
   TBody,
 } from "./BaseItemsTable.styled";
-import BaseTableItem, { type BaseTableItemProps } from "./BaseTableItem";
+import { Columns } from "./Columns";
+import type { ResponsiveProps } from "./utils";
 
 export type SortingOptions = {
   sort_column: string;
   sort_direction: "asc" | "desc";
 };
 
-interface SortableColumnHeaderProps
-  extends PropsWithChildren<Partial<HTMLAttributes<HTMLDivElement>>> {
-  name: string;
-  sortingOptions: SortingOptions;
-  onSortingOptionsChange: (newSortingOptions: SortingOptions) => void;
-}
+export type SortableColumnHeaderProps = {
+  name?: string;
+  sortingOptions?: SortingOptions;
+  onSortingOptionsChange?: (newSortingOptions: SortingOptions) => void;
+  isSortable?: boolean;
+} & PropsWithChildren<Partial<HTMLAttributes<HTMLDivElement>>>;
 
 export enum Sort {
   Asc = "asc",
   Desc = "desc",
 }
 
-const SortableColumnHeader = ({
-  name,
-  sortingOptions,
+export const SortableColumnHeader = ({
+  name = "",
+  sortingOptions = {
+    sort_column: "",
+    sort_direction: Sort.Asc,
+  },
   onSortingOptionsChange,
+  isSortable = !!onSortingOptionsChange,
   children,
+  hideAtContainerBreakpoint,
+  containerName,
   ...props
-}: SortableColumnHeaderProps) => {
+}: SortableColumnHeaderProps & ResponsiveProps) => {
   const isSortingThisColumn = sortingOptions.sort_column === name;
   const direction = isSortingThisColumn
     ? sortingOptions.sort_direction
     : Sort.Desc;
 
   const onSortingControlClick = () => {
     const nextDirection = direction === Sort.Asc ? Sort.Desc : Sort.Asc;
-    onSortingOptionsChange({
+    onSortingOptionsChange?.({
       sort_column: name,
       sort_direction: nextDirection,
     });
   };
 
   return (
-    <ColumnHeader>
+    <ColumnHeader
+      hideAtContainerBreakpoint={hideAtContainerBreakpoint}
+      containerName={containerName}
+    >
```

Comment:
On this branch, columns are hidden/shown depending on the width of the container. Rather than use the previously existing `col, th, td { &:nth-of-type ...` pattern, we add a `@container` query to certain cells in the table. This query sets `display: none` if the container shrinks below a given breakpoint.